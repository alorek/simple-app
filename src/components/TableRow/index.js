import React, {Component} from 'react';
import { Link } from 'react-router-dom'

class TableRow extends Component {
    render() {
        const recordInfo = this.props.data || {};
        const isViewable = (recordInfo).viewable === 'true';
        const link = `/record/${recordInfo.id}`;

        return (
            <tr key={recordInfo.id}>
                <td>{recordInfo.resource_type}</td>
                <td>{recordInfo.display_title}</td>
                <td>{recordInfo.media_type}</td>
                <td>{recordInfo.language}</td>
                <td>{isViewable ? <Link to={link} ><span className="glyphicon glyphicon-eye-open"></span></Link> : null}</td>
            </tr>
        );
    }
}

export default TableRow;
