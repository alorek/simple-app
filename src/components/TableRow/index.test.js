import React from 'react';
import renderer from 'react-test-renderer';

import TableRow from './index';

test('TableRow renders correctly', () => {

    const tree = renderer.create(
        <TableRow />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
