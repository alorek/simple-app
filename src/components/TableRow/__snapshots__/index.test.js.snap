// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow renders correctly 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
  <td />
</tr>
`;
